engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -1258,6 +1294,10 @@ public StreamingContainerAgent assignContainer(ContainerResource resource, InetS
       return null;
     }
 
+    //get groupid if container is deployed as part of redeployemnt process
+    if (redeployRequests.containsKey(container.getExternalId())) {
```

Comment:
When container is restarted old container is removed and new one is added. To track newly added container w.t.r. removal of old container I am doing this. Let me add some comments for clarity.
It may stay around but never be referred as we got new container id now and if restart happens that will be with new id. I will still make sure we remove this entry from somewhere.